extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracePropagator.java

```diff
@@ -108,14 +109,16 @@ public <C> Context extract(Context context, @Nullable C carrier, TextMapGetter<C
     if (carrier != null) {
       BaggageBuilder baggageBuilder = Baggage.builder();
       for (String key : getter.keys(carrier)) {
-        if (!key.startsWith(PREFIX_BAGGAGE_HEADER)) {
+        String lowercaseKey = key.toLowerCase(Locale.ROOT);
+        if (!lowercaseKey.startsWith(PREFIX_BAGGAGE_HEADER)) {
```

Comment:
alternatively could use
```java
if (!key.regionMatches(true, 0, PREFIX_BAGGAGE_HEADER, 0, PREFIX_BAGGAGE_HEADER.length())) {
  continue;
}
```
to avoid converting the whole key

Reply:
If we decided to keep the case of header, I will also apply this change.